### ISSUE TITLE
CanBeNonNullable false positive

### ISSUE THREAD

**Issue by eygraber** [detekt/detekt#7420]
## Expected Behavior
The `Locale` in this code shouldn't be flagged as `CanBeNonNullable`:

```kotlin
private fun setMessageTimestamp(
    messageType: MessageType,
    message: Message,
    locale: Locale?,
  ) {
    if(messageType == MessageType.ChecklistItem) {
      messageTimestamp?.text = "Marked completed"
    }
    else {
      val zonedDateTime = message.sentAt
      if(zonedDateTime != null && locale != null) {
        val dateFormatter = DataFormatter.instance.dateTimeFormatter(locale)
        val localDateTime = ZonedDateTime.ofInstant(zonedDateTime.toInstant(), ZoneId.systemDefault())
        messageTimestamp?.text = localDateTime.format(dateFormatter)
      }
    }
  }
```

## Observed Behavior
It is flagged

## Your Environment
* Version of detekt used: 1.23.6

**Comment by 3flex** [comment]
Can you please run your detekt task with debug enabled? There are likely inputs missing from the detekt task's classpath.

This should not be an issue for compilation analysis tasks in 2.0.0 due to improvements in the way tasks are setup by detekt Gradle plugin but I'd like to see the results first.

**Comment by efenderbosch-atg** [comment]
I have the same issue. Just used `@Suppress` until now. Here's my debug log:

```
❯ ./gradlew detektMain

> Task :detektMain
Loaded extensions: 
    
Registered rule sets: 
    io.gitlab.arturbosch.detekt.rules.bugs.PotentialBugProvider@17a4e8a8
    io.gitlab.arturbosch.detekt.rules.complexity.ComplexityProvider@24fd996c
    io.gitlab.arturbosch.detekt.rules.coroutines.CoroutinesProvider@31f51b37
    io.gitlab.arturbosch.detekt.rules.documentation.CommentSmellProvider@779c17fd
    io.gitlab.arturbosch.detekt.rules.empty.EmptyCodeProvider@47cd500e
    io.gitlab.arturbosch.detekt.rules.exceptions.ExceptionsProvider@f338762
    io.gitlab.arturbosch.detekt.rules.naming.NamingProvider@16060910
    io.gitlab.arturbosch.detekt.rules.performance.PerformanceProvider@1b2d7d2
    io.gitlab.arturbosch.detekt.rules.style.StyleGuideProvider@2c52aaca
    io.gitlab.arturbosch.detekt.formatting.FormattingProvider@3fc930b5
warning: classpath entry points to a non-existent location: /home/me/IdeaProjects/mono-repo/service/build/classes/groovy/main
Loaded extensions: 
    io.gitlab.arturbosch.detekt.rules.documentation.LicenceHeaderLoaderExtension@5346dae8
    io.github.detekt.metrics.processors.ProjectSLOCProcessor@55efd9b2
    io.github.detekt.metrics.processors.ProjectLOCProcessor@40490b77
    io.github.detekt.metrics.processors.ProjectCLOCProcessor@9031762
    io.github.detekt.metrics.processors.ProjectLLOCProcessor@6c770d30
    io.github.detekt.metrics.processors.ProjectCognitiveComplexityProcessor@47fe6b45
    io.github.detekt.metrics.processors.ProjectComplexityProcessor@4f64d27e
    io.github.detekt.metrics.processors.PropertyCountProcessor@5e4d64f7
    io.github.detekt.metrics.processors.FunctionCountProcessor@a87ec83
    io.github.detekt.metrics.processors.ClassCountProcessor@3e170ad5
    io.github.detekt.metrics.processors.PackageCountProcessor@5789d84b
    io.github.detekt.metrics.processors.KtFileCountProcessor@62566491
Registered rule sets: 
    io.gitlab.arturbosch.detekt.rules.bugs.PotentialBugProvider@1c7f8adb
    io.gitlab.arturbosch.detekt.rules.complexity.ComplexityProvider@2eda483b
    io.gitlab.arturbosch.detekt.rules.coroutines.CoroutinesProvider@64826d48
    io.gitlab.arturbosch.detekt.rules.documentation.CommentSmellProvider@134e1d64
    io.gitlab.arturbosch.detekt.rules.empty.EmptyCodeProvider@73876484
    io.gitlab.arturbosch.detekt.rules.exceptions.ExceptionsProvider@f60715f
    io.gitlab.arturbosch.detekt.rules.naming.NamingProvider@78ffa5d6
    io.gitlab.arturbosch.detekt.rules.performance.PerformanceProvider@1b392290
    io.gitlab.arturbosch.detekt.rules.style.StyleGuideProvider@5352d637
    io.gitlab.arturbosch.detekt.formatting.FormattingProvider@5f34bf8b
Loaded extensions: 
    io.gitlab.arturbosch.detekt.core.baseline.BaselineResultMapping@79bbc227
Loaded extensions: 
    io.github.detekt.report.sarif.SarifOutputReport@503cba13
    io.github.detekt.report.xml.XmlOutputReport@62fc31fe
    io.github.detekt.report.txt.TxtOutputReport@62ae153b
    io.github.detekt.report.html.HtmlOutputReport@731e3a07
    io.github.detekt.report.md.MdOutputReport@4647fa55
Loaded extensions: 
    io.gitlab.arturbosch.detekt.core.reporting.console.LiteFindingsReport@15466dd
/home/me/IdeaProjects/monorepo/service/src/main/kotlin/com/example/a2o/api/auction/AuctionService.kt:64:9: The nullable parameter 'maxLotUpdatedDate' can be made non-nullable. [CanBeNonNullable]

Phase LoadConfig took 25ms
Phase CreateSettings took 0s
Phase ValidateConfig took 11ms
Phase Parsing took 299ms
Phase Binding took 4.326s
Phase LoadingExtensions took 1ms
Phase Analyzer took 5.903s
Phase Reporting took 16ms

> Task :detektMain FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':detektMain'.
> Analysis failed with 1 weighted issues.
```

**Comment by eygraber** [comment]
```
> Task :app:detektDebug
Loaded extensions: 
    
Registered rule sets: 
    io.gitlab.arturbosch.detekt.rules.bugs.PotentialBugProvider@f4a4d18
    io.gitlab.arturbosch.detekt.rules.complexity.ComplexityProvider@361efa61
    io.gitlab.arturbosch.detekt.rules.coroutines.CoroutinesProvider@72b6d77
    io.gitlab.arturbosch.detekt.rules.documentation.CommentSmellProvider@533bc5d1
    io.gitlab.arturbosch.detekt.rules.empty.EmptyCodeProvider@ec46023
    io.gitlab.arturbosch.detekt.rules.exceptions.ExceptionsProvider@4f484e68
    io.gitlab.arturbosch.detekt.rules.naming.NamingProvider@13d4c223
    io.gitlab.arturbosch.detekt.rules.performance.PerformanceProvider@1e4c98c5
    io.gitlab.arturbosch.detekt.rules.style.StyleGuideProvider@33243657
    io.nlopez.compose.rules.detekt.ComposeRuleSetProvider@3e920555
    com.eygraber.detekt.rules.formatting.FormattingRuleSetProvider@be20d7e
    com.eygraber.detekt.rules.style.StyleRuleSetProvider@71923677
Loaded extensions: 
    io.gitlab.arturbosch.detekt.rules.documentation.LicenceHeaderLoaderExtension@32be2e7f
    io.github.detekt.metrics.processors.ProjectSLOCProcessor@42614a67
    io.github.detekt.metrics.processors.ProjectLOCProcessor@5ad4ae08
    io.github.detekt.metrics.processors.ProjectCLOCProcessor@130c9768
    io.github.detekt.metrics.processors.ProjectLLOCProcessor@50829ea0
    io.github.detekt.metrics.processors.ProjectCognitiveComplexityProcessor@18f8c4a5
    io.github.detekt.metrics.processors.ProjectComplexityProcessor@67a39d5c
    io.github.detekt.metrics.processors.PropertyCountProcessor@10e3f117
    io.github.detekt.metrics.processors.FunctionCountProcessor@54500a35
    io.github.detekt.metrics.processors.ClassCountProcessor@ad331f6
    io.github.detekt.metrics.processors.PackageCountProcessor@71e1048d
    io.github.detekt.metrics.processors.KtFileCountProcessor@7e0528ef
Registered rule sets: 
    io.gitlab.arturbosch.detekt.rules.bugs.PotentialBugProvider@301d4427
    io.gitlab.arturbosch.detekt.rules.complexity.ComplexityProvider@5f3a6caf
    io.gitlab.arturbosch.detekt.rules.coroutines.CoroutinesProvider@4faecef3
    io.gitlab.arturbosch.detekt.rules.documentation.CommentSmellProvider@cac5ec7
    io.gitlab.arturbosch.detekt.rules.empty.EmptyCodeProvider@56ca0f09
    io.gitlab.arturbosch.detekt.rules.exceptions.ExceptionsProvider@5a195e6b
    io.gitlab.arturbosch.detekt.rules.naming.NamingProvider@63f9da8d
    io.gitlab.arturbosch.detekt.rules.performance.PerformanceProvider@d4dea53
    io.gitlab.arturbosch.detekt.rules.style.StyleGuideProvider@b790eb4
    io.nlopez.compose.rules.detekt.ComposeRuleSetProvider@1b03e3b6
    com.eygraber.detekt.rules.formatting.FormattingRuleSetProvider@36836fe9
    com.eygraber.detekt.rules.style.StyleRuleSetProvider@547e0418
Loaded extensions: 
    io.gitlab.arturbosch.detekt.core.baseline.BaselineResultMapping@7cec0ba6
Loaded extensions: 
    io.github.detekt.report.sarif.SarifOutputReport@69c0bd68
    io.github.detekt.report.xml.XmlOutputReport@61eb56a9
    io.github.detekt.report.txt.TxtOutputReport@4ff6af78
    io.github.detekt.report.html.HtmlOutputReport@7ea3c5f1
    io.github.detekt.report.md.MdOutputReport@12640cb0
Loaded extensions: 
    io.gitlab.arturbosch.detekt.core.reporting.console.LiteFindingsReport@2e823c38
    io.gitlab.arturbosch.detekt.core.reporting.console.FindingsReport@cb14c5
/home/eli/workspace/myApp/app/src/main/kotlin/com/example/messages/ui/MessageViewHolder.kt:172:5: The nullable parameter 'locale' can be made non-nullable. [CanBeNonNullable]

style - 10min debt
        CanBeNonNullable - [The nullable parameter 'locale' can be made non-nullable.] at /home/eli/workspace/myApp/app/src/main/kotlin/com/example/messages/ui/MessageViewHolder.kt:172:5

Overall debt: 10min

Phase LoadConfig took 2ms
Phase CreateSettings took 0s
Phase ValidateConfig took 7ms
Phase Parsing took 125ms
Phase Binding took 4.483s
Phase LoadingExtensions took 1ms
Phase Analyzer took 392ms
Phase Reporting took 8ms

> Task :app:detektDebug FAILED
```

**Comment by efenderbosch-atg** [comment]
This does appear to be fixed in the `1.23.7` pre-release

**Comment by eygraber** [comment]
Ping

**Comment by efenderbosch-atg** [comment]
Bumb for the bot.

**Comment by mulledwineapps** [comment]
Is there any news? I'm encountering the same issue.
```
private fun example(param: Any?) {
    param?.let {
        // some logic
    }
}
```

**Comment by BraisGabin** [comment]
That's not a false positive. You don't need to pass `Any?`. You could pass `Any` because this function does nothing when the value is `null`.

**Comment by eygraber** [comment]
Can this get labeled as `never gets stale`?
